stock_picking_declared_value/models/stock_picking.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2025 Nitrokey GmbH
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    # Add a computed field to show the total declared value of the picking
+    amount_total = fields.Monetary(
+        string="Total Declared Value",
+        compute="_compute_amount_total",
+        store=True,
+        help="Total declared value of the picking based on the values of the products",
+    )
+    currency_id = fields.Many2one(
+        "res.currency",
+        string="Currency",
+        related="sale_id.currency_id",
```

Comment:
If there is no `sale.order` linked with the picking the currency is False but still it's still used. Turn it into a computed field based on `sale_id` and default to the currency on the company.